enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,587 @@
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.storeEndpoint.serviceConfig.region`
+- `hostedcluster.v1beta1.hypershift.openshift.io.spec.etcd.backup.storeEndpoint.serviceConfig.tags`
+
+Main security points:
+
+- Same secret management as being carried out by the HO, CPO, and HCCO.
+- No additional permissions needed to perform and manage the backups.
+- We will segregate permissions to create the secret in the HostedCluster namespace and replicate those secrets to the HostedControlPlane namespace by the controller.
+- The [persona](https://hypershift-docs.netlify.app/reference/concepts-and-personas/) that will benefit from this new API will be the Cluster Service Provider and the Cluster Service Consumer.
+- Like other endpoints in Hypershift, the inputs are restricted by the defined types:
+  - Enums are limited to the types defined in the code.
+  - Strings are restricted to a defined number of characters via OpenAPI.
+  - Integers are defined by the Go language.
+- All mentioned endpoints require authentication.
+- The backup performed is encrypted by default using the key generated for that ETCD cluster.
+- The permissions of the provided credentials could be very limited, only allowing for pushing the backup binary to a specific S3 bucket. Listing, modifying, or similar actions are not necessary.
+- This new component has not yet undergone DUST testing or penetration testing.
+
+**How will security be reviewed and by whom?**
+
+From RH Security team
+
+- David Sastre (@dawud) (Security Architect dedicated to Hypershift)
+- Florencio Cano (Backup)
+
+From the internal team:
+
+- @derekwaynecarr
+- @csrwng
+- @enxebre
+
+**How will UX be reviewed and by whom?**
+
+- @hypershift-team
+
+### Drawbacks
+
+N/A
+
+## Open Questions
+
+- Do we need a retryPolicy for the failure processes?
+- The Garbage collector will also clean the remote backups?
+- Could the use of backups affect performance at scale, with 80 HostedClusters performing backups simultaneously?
+
+## Test Plan
+
+### Unit Testing
+
+Like any other component, every part of the controller that makes sense to be tested will include unit testing.
+
+### Conformance Testing
+
+Each of the controllers, `ETCDBackupController`, and `ETCDUploadController`, will have its own section to test the behavior of each aspect.
+
+For `ETCDBackupController`:
+
+- Test one-shot backup and validate
+- Test scheduled backup and validate
+
+For `ETCDUploadController`:
+
+- Test one-shot backup, validate and upload
+- Test scheduled backup, validate and upload
+
+### E2E
+
+For E2E testing, the plan is to develop tests covering the final backup functionality. Currently, tests are available only for AWS, which will be the first provider to be covered. Eventually, more providers will be included. For the Agent/Kubevirt providers, it is likely that we will use Minio or a similar service to mimic the S3 API.
```

Comment:
i believe ODF has an object storage solution

Reply:
Yep in the past it was called Nooba, i think it's still there and it allow block storage. I think for the first iteration we can keep a MinIO approach because in the end it uses S3 like API. And focus in a possible approach for ODF block storage API.